src/platform_impl/macos/event.rs

```diff
@@ -89,6 +89,106 @@ pub fn char_to_keycode(c: char) -> Option<VirtualKeyCode> {
+        VirtualKeyCode::N => 'n',
+        VirtualKeyCode::O if m.shift() => 'O',
+        VirtualKeyCode::O => 'o',
+        VirtualKeyCode::P if m.shift() => 'P',
+        VirtualKeyCode::P => 'p',
+        VirtualKeyCode::Q if m.shift() => 'Q',
+        VirtualKeyCode::Q => 'q',
+        VirtualKeyCode::R if m.shift() => 'R',
+        VirtualKeyCode::R => 'r',
+        VirtualKeyCode::S if m.shift() => 'S',
+        VirtualKeyCode::S => 's',
+        VirtualKeyCode::T if m.shift() => 'T',
+        VirtualKeyCode::T => 't',
+        VirtualKeyCode::U if m.shift() => 'U',
+        VirtualKeyCode::U => 'u',
+        VirtualKeyCode::V if m.shift() => 'V',
+        VirtualKeyCode::V => 'v',
+        VirtualKeyCode::W if m.shift() => 'W',
+        VirtualKeyCode::W => 'w',
+        VirtualKeyCode::X if m.shift() => 'X',
+        VirtualKeyCode::X => 'x',
+        VirtualKeyCode::Y if m.shift() => 'Y',
+        VirtualKeyCode::Y => 'y',
+        VirtualKeyCode::Z if m.shift() => 'Z',
+        VirtualKeyCode::Z => 'z',
+        VirtualKeyCode::Key1 if m.shift() => '!',
+        VirtualKeyCode::Key1 => '1',
+        VirtualKeyCode::Key2 if m.shift() => '@',
+        VirtualKeyCode::Key2 => '2',
+        VirtualKeyCode::Key3 if m.shift() => '#',
+        VirtualKeyCode::Key3 => '3',
+        VirtualKeyCode::Key4 if m.shift() => '$',
+        VirtualKeyCode::Key4 => '4',
+        VirtualKeyCode::Key5 if m.shift() => '%',
+        VirtualKeyCode::Key5 => '5',
+        VirtualKeyCode::Key6 if m.shift() => '^',
+        VirtualKeyCode::Key6 => '6',
+        VirtualKeyCode::Key7 if m.shift() => '&',
+        VirtualKeyCode::Key7 => '7',
+        VirtualKeyCode::Key8 if m.shift() => '*',
+        VirtualKeyCode::Key8 => '8',
+        VirtualKeyCode::Key9 if m.shift() => '(',
+        VirtualKeyCode::Key9 => '9',
+        VirtualKeyCode::Key0 if m.shift() => ')',
+        VirtualKeyCode::Key0 => '0',
+        VirtualKeyCode::Minus if m.shift() => '_',
+        VirtualKeyCode::Minus => '-',
+        VirtualKeyCode::Equals if m.shift() => '+',
+        VirtualKeyCode::Equals => '=',
+        VirtualKeyCode::LBracket if m.shift() => '{',
+        VirtualKeyCode::LBracket => '[',
+        VirtualKeyCode::RBracket if m.shift() => '}',
+        VirtualKeyCode::RBracket => ']',
+        VirtualKeyCode::Backslash if m.shift() => '|',
+        VirtualKeyCode::Backslash => '\\',
+        VirtualKeyCode::Semicolon if m.shift() => ':',
+        VirtualKeyCode::Semicolon => ';',
+        VirtualKeyCode::Apostrophe if m.shift() => '"',
+        VirtualKeyCode::Apostrophe => '\'',
+        VirtualKeyCode::Comma if m.shift() => '<',
+        VirtualKeyCode::Comma => ',',
+        VirtualKeyCode::Period if m.shift() => '>',
+        VirtualKeyCode::Period => '.',
+        VirtualKeyCode::Slash if m.shift() => '?',
+        VirtualKeyCode::Slash => '/',
+        VirtualKeyCode::Grave if m.shift() => '~',
+        VirtualKeyCode::Grave => '`',
+        _ => return None,
```

Comment:
This is not the right way to do this. You're assuming QUERTY here for US keyboard. However there're keyboard for which this isn't true. Hardcoding isn't an option.

Reply:
Yes, you are right, it is assuming QWERTY. I'm hoping to get attention to the issues Mac users face.